plan/physical_plan_builder.go

```diff
@@ -354,7 +354,7 @@ func (ds *DataSource) convertToIndexScan(prop *requiredProp, idx *model.IndexInf
 			}
 		}
 	}
-	if matchProperty && prop.expectedCnt < math.MaxFloat64 {
+	if matchProperty && prop.expectedCnt < math.MaxFloat64 && prop.expectedCnt > ds.statsAfterSelect.count {
```

Comment:
I think that it is fine for `expectCnt` not larger than `statsAfterSelect.count`.